diff --git a/dist/shared/binding-s-V_wTpj.mjs b/dist/shared/binding-s-V_wTpj.mjs
index 01316e9c93e935130c7c2f2fdf607e695e2bfa6c..487e9bffdd4cf07521a452b803e4018ae4b1c0d7 100644
--- a/dist/shared/binding-s-V_wTpj.mjs
+++ b/dist/shared/binding-s-V_wTpj.mjs
@@ -48,18 +48,81 @@ var require_webcontainer_fallback = /* @__PURE__ */ __commonJSMin(((exports, mod
 //#endregion
 //#region src/binding.cjs
 var require_binding = /* @__PURE__ */ __commonJSMin(((exports, module) => {
-	const { readFileSync } = __require("node:fs");
+	const { readFileSync, openSync, readSync, close } = __require("node:fs");
 	let nativeBinding = null;
 	const loadErrors = [];
+  // Patch related to the following issues
+  // https://github.com/napi-rs/napi-rs/pull/3241
+  // https://github.com/napi-rs/napi-rs/issues/3223
+  // https://github.com/rolldown/rolldown/issues/9133
+  // https://github.com/lovell/detect-libc/issues/29
+  // https://github.com/lovell/detect-libc/commit/4d6eafecb2bd6c45f8a95822966cf7c08f79d21b (fix copied from here)
 	const isMusl = () => {
 		let musl = false;
 		if (process.platform === "linux") {
-			musl = isMuslFromFilesystem();
+			musl = isMuslFromInterpreter();
+			if (musl === null) musl = isMuslFromFilesystem();
 			if (musl === null) musl = isMuslFromReport();
 			if (musl === null) musl = isMuslFromChildProcess();
 		}
 		return musl;
 	};
+  const isMuslFromInterpreter = () => {
+    try {
+      const SELF_PATH = '/proc/self/exe';
+      const MAX_LENGTH = 2048;
+      const fd = openSync(SELF_PATH, 'r');
+      const buffer = Buffer.alloc(MAX_LENGTH);
+      const bytesRead = readSync(fd, buffer, 0, MAX_LENGTH, 0);
+      close(fd);
+      const selfContent = buffer.subarray(0, bytesRead);
+      const path = interpreterPath(selfContent);
+      return isMuslFromInterpreterPath(path);
+    } catch (e) {}
+    return null;
+  }
+
+  const interpreterPath = (elf) => {
+    if (elf.length < 64) {
+      return null;
+    }
+    if (elf.readUInt32BE(0) !== 0x7F454C46) {
+      // Unexpected magic bytes
+      return null;
+    }
+    if (elf.readUInt8(4) !== 2) {
+      // Not a 64-bit ELF
+      return null;
+    }
+    if (elf.readUInt8(5) !== 1) {
+      // Not little-endian
+      return null;
+    }
+    const offset = elf.readUInt32LE(32);
+    const size = elf.readUInt16LE(54);
+    const count = elf.readUInt16LE(56);
+    for (let i = 0; i < count; i++) {
+      const headerOffset = offset + (i * size);
+      const type = elf.readUInt32LE(headerOffset);
+      if (type === 3) {
+        const fileOffset = elf.readUInt32LE(headerOffset + 8);
+        const fileSize = elf.readUInt32LE(headerOffset + 32);
+        return elf.subarray(fileOffset, fileOffset + fileSize).toString().replace(/\0.*$/g, '');
+      }
+    }
+    return null;
+  };
+
+  const isMuslFromInterpreterPath = (path) => {
+    if (path) {
+      if (path.includes('/ld-musl-')) {
+        return true;
+      } else if (path.includes('/ld-linux-')) {
+        return false;
+      }
+    }
+    return null;
+  };
 	const isFileMusl = (f) => f.includes("libc.musl-") || f.includes("ld-musl-");
 	const isMuslFromFilesystem = () => {
 		try {
